components/cards/UserStatus.tsx

```diff
@@ -74,11 +77,11 @@ export default function UserStatus({
         style={[styles.animationContainer, style, cardAnimatedStyle]}
         {...rest}>
         <View style={styles.userContainer}>
-          {user && (
+          {user && signal && (
             <UserAvailability
               fullName={user.names}
-              time={timeSlot}
-              activity={activity}
+              time={signal.when}
+              activity={signal.status_message}
```

Comment:
shouldn't we keep using data from the context since the `useFetchMySignal` updates it. and in other places where we need these data we need the values from the context "temporaryStatus"

Reply:
I used data directly from the useQuery response to facilitate optimistic update and in all the other places that i need to invalidate the query.